### PR TITLE
EL-485 - remove housing costs cap if applicant has a partner

### PR DESCRIPTION
**Pull request by starswan**
https://dsdmoj.atlassian.net/browse/EL-485

Introduce rule that specifies that the housing cost cap is removed for both calculations if client has a partner

### CODE REVIEW

File: app/services/workflows/non_passported_workflow.rb [hunk 1]
```diff
@@ -53,17 +53,66 @@ def collate_and_assess_gross_income
       end
     end
 
+    class PersonDecorator
```
Comment (patrick-laa) [inline app/services/workflows/non_passported_workflow.rb:56]: `Decorator` in this codebase is used a lot, and up until now uniformly to name classes you can call `as_json` on. I know that this class is a decorator in the literal sense, but is there another name we can call it given that Decorator has this quite specific meaning?

**Review by patrick-laa — COMMENTED** [review]
One qualm about naming (happy to be argued out of it); everything else looks great.

File: app/services/collators/childcare_collator.rb [hunk 2]
```diff
@@ -31,7 +30,7 @@ def monthly_child_care_cash
     end
 
     def applicant_has_dependant_child?
```
Comment (patrick-laa) [inline app/services/collators/childcare_collator.rb:32]: Oh, this should be `person_has_dependant_child?` and likewise `applicant_is_employed?` and `applicant_has_student_loan?` should be renamed

File: app/services/workflows/non_passported_workflow.rb (hunk 1 above)
Comment (starswan) [reply app/services/workflows/non_passported_workflow.rb:56]: +1 - Actually this isn't a decorator either as it doesn't maintain the interface of the decorated object.

**Review by patrick-laa — APPROVED** [review]

**Pull request merged**